lib/src/json_serializable_generator.dart

```diff
@@ -221,14 +227,58 @@ class JsonSerializableGenerator
         // ...resetting `isList` to `false`.
         isList = false;
       }
-    }
 
-    if (!isList && _coreIterableChecker.isAssignableFromType(fieldType)) {
-      // Then we need to add `?.toList()
-      expression += "?.toList()";
+      if (!isList) {
+        // Then we need to add `?.toList()
```

Comment:
avoid comments that only repeat what the code is doing - if you want to keep a comment here it should say why we need this.